temporal-sdk/src/main/java/io/temporal/internal/nexus/PayloadSerializer.java

```diff
@@ -52,7 +53,14 @@ public Content serialize(@Nullable Object o) {
   public @Nullable Object deserialize(Content content, Type type) {
     try {
       Payload payload = Payload.parseFrom(content.getData());
-      return dataConverter.fromPayload(payload, type.getClass(), type);
+      if ((type instanceof Class)) {
```

Comment:
This is very similar logic to how jackson works https://github.com/FasterXML/jackson-databind/blob/31f1e45f7414a1ac162feebc1e657afb8914d1b3/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java#L1450 there are some cases missing because we don't allow generic services

Reply:
This makes sense to me, however you might want to check that `getRawType()` is a `Class` explicitly unless you want the more cryptic Java casting exception. Technically a parameterized type could be a wildcard or type variable (though hopefully our checks at definition time prevent type variables).

Reply:
Actually, I may have had that backwards, the raw/outer type is likely not a non-class (not even sure it can be, you can't have `T<U>` or `?<U>`). It's the argument types I was concerned about, but that can be left to the converter (i.e. Jackson) to figure out whether they can accept.